contrib/mesos/pkg/scheduler/service/util.go

```diff
@@ -86,3 +94,131 @@ func hashExecutorInfo(info *mesos.ExecutorInfo) uint64 {
 	table := crc64.MakeTable(crc64.ECMA)
 	return crc64.Checksum(buf.Bytes(), table)
 }
+
+type defaultFunc func(pod *api.Pod) error
+
+// tryDecodeSinglePod was copied from pkg/kubelet/config/common.go v1.0.5
+func tryDecodeSinglePod(data []byte, defaultFn defaultFunc) (parsed bool, pod *api.Pod, err error) {
+	// JSON is valid YAML, so this should work for everything.
+	json, err := utilyaml.ToJSON(data)
+	if err != nil {
+		return false, nil, err
+	}
+	obj, err := api.Scheme.Decode(json)
+	if err != nil {
+		return false, pod, err
+	}
+	// Check whether the object could be converted to single pod.
+	if _, ok := obj.(*api.Pod); !ok {
+		err = fmt.Errorf("invalid pod: %+v", obj)
+		return false, pod, err
+	}
+	newPod := obj.(*api.Pod)
+	// Apply default values and validate the pod.
+	if err = defaultFn(newPod); err != nil {
+		return true, pod, err
+	}
+	if errs := validation.ValidatePod(newPod); len(errs) > 0 {
+		err = fmt.Errorf("invalid pod: %v", errs)
+		return true, pod, err
+	}
+	return true, newPod, nil
+}
+
+type filterFunc func(pod *api.Pod) (bool, error)
+
+func (filter filterFunc) podEntries(in <-chan *archive.Entry) <-chan *archive.Entry {
+	out := make(chan *archive.Entry)
+	go func() {
+		defer close(out)
+		for entry := range in {
+			if entry.Dir {
+				continue
+			}
+			if entry.Err != nil {
+				log.Errorf("error fetching static pod file %q: %v", entry.Path, entry.Err)
+				continue
+			}
+			keep := false
+			defaultFn := func(pod *api.Pod) (err error) {
+				keep, err = filter(pod)
+				return
+			}
+			parsed, pod, err := tryDecodeSinglePod(entry.Data, defaultFn)
+			if !parsed {
+				if err != nil {
+					entry.Err = err
+					log.Errorf("error parsing static pod file %q: %v", entry.Path, err)
+				}
+				continue
+			}
+			if err != nil {
+				entry.Err = err
+				log.Errorf("error validating static pod file %q: %v", entry.Path, err)
+				continue
+			}
+			if keep {
+				// marshal back into JSON form to pick up any changes made to
+				// the pod specification
+				updated, err := api.Scheme.EncodeToVersion(pod, "v1")
+				if err != nil {
+					entry.Err = err
+					log.Errorf("error serializing pod json for %v/%v: %v", pod.Namespace, pod.Name, err)
+					continue
+				}
+				entry.Data = updated
+				out <- entry
+			}
+		}
+	}()
+	return out
+}
+
+func makeStaticPodValidator(limitCPU resource.CPUShares, limitMem resource.MegaBytes, accumCPU, accumMem *float64, minimalResources bool) filterFunc {
```

Comment:
Having this code in util.go is not nice. Quoting Tomas: put code into util.go is a sign of bad concepts. At least move it into its own file staticpods.go.